cmse/cmse.md

```diff
@@ -1138,9 +1145,45 @@ caller stack frame](#figure5).
 
 ![<span id="figure5" class="citation-label">**Entry function's caller stack frame**</span>](stack-frame-entry.svg)
 
-### Return from an entry function
+Calls from Non-secure state follow the [[AAPCS]](#AAPCS), which states that the
+caller is responsible for zero- or sign-extending arguments of integral
+Fundamental Data Types smaller than a word to 4 bytes. An Entry function must
+not assume that callers follow this rule, that is, it cannot presume that
+integral parameters will have been zero- or sign-extended to 4 bytes. For
+example, an attacker might create code that passes arguments out of their
+declared type's range in an attempt to cause out-of-bounds memory accesses.
 
 <span id="requirement-47" class="requirement-box"></span>
+> A compiler generating code for an entry function must, for each parameter
+> that is an integral Fundamental Data Type smaller than a word, narrow the
+> parameter to a value within its declared type's range.
```

Comment:
Are there other ways to "narrow" the parameter other than sign- or zero- extending it? If not, I would suggest rephrasing in terms of sign-extending/zero-extending to 4-bytes, as it looks more clear.

Reply:
In discussions held previously, it was mentioned that saying truncate+extend explicitly could be overprescriptive. A compiler could for example decide for a check-and-abort implementation instead, which would be as valid.

Reply:
But in line with one of @ostannard's comments, one can argue that the current form may already be overprescriptive.